src/shared/test/frame-error-capture-test.js

```diff
@@ -0,0 +1,144 @@
+import { delay } from '../../test-util/wait';
+import {
+  captureErrors,
+  handleErrorsInFrames,
+  sendError,
+  sendErrorsTo,
+} from '../frame-error-capture';
+
+describe('shared/frame-error-capture', () => {
+  let errorEvents;
+
+  function handleMessage(event) {
+    if (event.data.type === 'hypothesis-error') {
+      errorEvents.push(event.data);
+    }
+  }
+
+  beforeEach(() => {
+    errorEvents = [];
+    window.addEventListener('message', handleMessage);
+  });
+
+  afterEach(() => {
+    window.removeEventListener('message', handleMessage);
+    sendErrorsTo(null);
+  });
+
+  describe('captureErrors', () => {
+    it('returns wrapped callback', () => {
+      const callback = captureErrors((a, b) => a * b, 'Testing captureErrors');
+      const result = callback(7, 8);
+      assert.equal(result, 56);
+    });
+
+    it('captures errors and forwards them to the current handler frame', async () => {
+      sendErrorsTo(window);
+
+      const error = new Error('Test error');
+      const callback = captureErrors(() => {
+        throw error;
+      }, 'Testing captureErrors');
+
+      assert.throws(() => {
+        callback();
+      }, 'Test error');
+
+      await delay(0);
+
+      assert.equal(errorEvents.length, 1);
+      assert.match(errorEvents[0], {
+        context: 'Testing captureErrors',
+        error: sinon.match({ message: 'Test error' }),
+        type: 'hypothesis-error',
+      });
+    });
+
+    it('does not forward errors if there is no handler frame', () => {
```

Comment:
This test belongs more to the `sendError` function.

Reply:
I agree. I've added a separate block of tests for `sendError`.